### PR TITLE
refs #16250 - mark advanced parameters

### PR DESCRIPTION
**Pull request by stbenjam**
I suppose it's debatable what's advanced or not, but I tried to mark as advanced everything but the most commonly used parameters.  This further reduces the size of foreman-installer --help.

### CODE REVIEW

**Comment by mmoll** [comment]
@stbenjam please rebase
:+1: in general

File: manifests/init.pp [hunk 1]
```diff
@@ -62,29 +130,6 @@
 # $plugin_version::             foreman plugins package version, it's passed to ensure parameter of package resource
 #                               can be set to 'installed', 'latest', 'present' only
 #
-# $db_manage::                  if enabled, will install and configure the database server on this host
-#                               type:boolean
-#
-# $db_type::                    Database 'production' type (valid types: mysql/postgresql/sqlite)
-#
-# $db_adapter::                 Database 'production' adapter
```
Comment (domcleal) [inline manifests/init.pp]: Not sure that the unmanaged database options should be moved, I'd suggest only keeping `$db_type` in the main parameters docs and _possibly_ db_manage, as type is the main parameter for changing the DB impl.

File: manifests/init.pp [hunk 2]
```diff
@@ -196,25 +218,6 @@
 # $loggers::                    Enable or disable specific loggers, e.g. {"sql" => true}
 #                               type:hash
 #
-# $email_conf::                 email configuration file, defaults to /etc/foreman/email.yaml
```
Comment (domcleal) [inline manifests/init.pp]: advanced, shouldn't need to be changed

File: manifests/init.pp [hunk 3]
```diff
@@ -196,25 +218,6 @@
...
-#
-# $email_source::               template to use for email configuration file
```
Comment (domcleal) [inline manifests/init.pp]: ditto

File: manifests/init.pp [hunk 4]
```diff
@@ -2,11 +2,79 @@
...
+#                               type:boolean
 #
 # $puppetrun::                  Should foreman be able to start puppetruns on nodes
 #                               type: boolean
 #
+# === Advanced parameters:
+#
+# $foreman_url::                URL on which foreman is going to run
+#
 # $unattended::                 Should foreman manage host provisioning as well
```
Comment (domcleal) [inline manifests/init.pp]: regular app setting, should probably not be in advanced

**Comment by domcleal** [comment]
Some of the repo settings, e.g. configure_epel_repo should perhaps not be in advanced too. It might be best to categorise settings (DB, email setting categories) rather than just have a simple/advanced block.

**Comment by stbenjam** [comment]
>  It might be best to categorise settings (DB, email setting categories) rather than just have a simple/advanced block.

The problem is I want some of those basic settings users often touch to be visible immediately in Kafo --help, it'd be weird to not have all of the e-mail settings under an Email section, right?

File: manifests/init.pp (hunk 4 above)
Comment (stbenjam) [inline manifests/init.pp]: Should it? I don't mind moving it, but in my mind, I put this is the same category as 'authorization' - these legacy components that you *can* turn off in Settings, but plugins or even core configuration may misbehave if you do it because we don't particularly test well in those situations.

**Comment by stbenjam** [comment]
Updated

File: manifests/init.pp [hunk 5]
```diff
@@ -2,11 +2,58 @@
...
+# $admin_last_name::            Last name of the initial admin user
+#
+# $admin_email::                E-mail address of the initial admin user
+#
+# $db_manage::                  if enabled, will install and configure the database server on this host
+#                               type:boolean
+#
+# $db_type::                    Database 'production' type (valid types: mysql/postgresql/sqlite)
+#
+# $email_source::               template to use for email configuration file
```
Comment (domcleal) [inline manifests/init.pp]: Move this to advanced, it can't be changed without adding a template to the environment.

File: manifests/init.pp [hunk 6]
```diff
@@ -2,11 +2,58 @@
...
+# $admin_last_name::            Last name of the initial admin user
+#
+# $admin_email::                E-mail address of the initial admin user
+#
+# $db_manage::                  if enabled, will install and configure the database server on this host
+#                               type:boolean
+#
+# $db_type::                    Database 'production' type (valid types: mysql/postgresql/sqlite)
+#
+# $email_delivery_method::      can be sendmail or smtp regarding to foreman documentation
```
Comment (ekohl) [inline manifests/init.pp]: Maybe rephrase this while you're at it?

**Comment by mmoll** [comment]
cherry-picked as 6a42cf2, with a little rewording to email_delivery_method

**Comment by mmoll** [comment]
thanks @stbenjam!
